draft-ietf-quic-version-negotiation.md

```diff
@@ -289,49 +322,30 @@ version that they initially attempted. Once a client has reacted to a Version
-the version in the long header that carried the handshake version information.
-Similarly, clients MUST validate that the server's `Negotiated Version` matches
-the long header version. If an endpoint's validation fails, it MUST close the
-connection. If the connection was using QUIC version 1, it MUST be closed with
-a transport error of type `VERSION_NEGOTIATION_ERROR`.
-
-When a server parses the client's handshake version information, if the
-`Received Negotiation Version Count` is not zero, the server MUST validate that
-it could have sent the Version Negotation packet described by the client in
-response to a first flight of version `Previously Attempted Version`. In
-particular, the server MUST ensure that there are no versions that it supports
-that are absent from the Received Negotiation Versions, and that the ordering
-matches the server's preference. If this validation fails, the server MUST
-close the connection. If the connection was using QUIC version 1, it MUST be
-closed with a transport error of type `VERSION_NEGOTIATION_ERROR`. This
-mitigates an attacker's ability to forge Version Negotiation packets to force a
-version downgrade.
-
-If a server operator is progressively deploying a new QUIC version throughout
-its fleet, it MAY perform a two-step process where it first progressively adds
-support for the new version, but without enforcing its presence in `Received
-Negotiation Versions`. Once all servers have been upgraded, the second step is
-to start enforcing that the new version is present in `Received Negotiation
-Versions`. This opens connections to version downgrades during the upgrade
-window, since those could be due to clients communicating with both upgraded
-and non-upgraded servers.
-
-If an endpoint receives its peer's Handshake Version Information and fails to
-parse it (for example, if it is too short), then the endpoint MUST close the
-connection. If the connection was using QUIC version 1, it MUST be closed with
-a transport error of type `TRANSPORT_PARAMETER_ERROR`.
-
-
-# Supported Versions
-
-The server's `Supported Version` list allows it to communicate the full list of
-versions it supports to the client. In the case where clients initially attempt
-connections with the oldest version they support, this allows them to be
-notified of more recent versions the server supports. If the client notices
-that the server supports a version that is more preferable that the one
-initially attempted by default, the client SHOULD cache that information and
-attempt the preferred version in subsequent connections.
+Both endpoints MUST parse their peer's Version Information during the
+handshake. If the Version Information was missing or if parsing it failed (for
+example, if it is too short or if its length is not divisible by four), then
+the endpoint MUST close the connection. If the connection was using QUIC
+version 1, it MUST be closed with a transport error of type
+`TRANSPORT_PARAMETER_ERROR`.
```

Comment:
Doesn't this mean you can't do incompatible VN when you offer VN=1?

Reply:
Clarified this by using a semi-colon.